radis/lbl/base.py

```diff
@@ -3245,7 +3283,14 @@ def _cutoff_linestrength(self, cutoff=None):
             ) from err
 
         # update df1:
-        self.df1 = pd.DataFrame(df[~b])
+        if self.dataframe_type == "pandas":
+            self.df1 = pd.DataFrame(df[~b])
+        elif self.dataframe_type == "vaex":
+            self.df1 = df[
+                ~b
+            ].materialize()  # If materialized is used time taken is reduced but memory efficiency is slightly reduced.
```

Comment:
@1someshverma @minouHub to my understanding of vaex, this is a massive move : you're removing all virtual columns here, and adding them in memory.

1. Is it necessary ? 
2. If so (because CPU-time is slow for instance) we might want to have different modes; with a focus on saving memory or saving CPU-time.

Reply:
It is not necessary, but it improves time performance for vaex. But we can consider adding mode as saving memory or saving CPU time.